R/expressions.R

```diff
@@ -166,10 +166,20 @@ setMethod("!", "CrunchExpr", function (x) zfuncExpr("not", x))
         length(table) > 2 &&
         identical(as.numeric(head(table, 1):tail(table, 1)), as.numeric(table))) {
 
+        # set beg(ining) and end appropriately in case we have been given a rev
+        # sequence (eg 20:1) ZCL returns nothing if asked for between 20 and 1
+        if (head(table, 1) < tail(table, 1)) {
+            beg <- head(table, 1)
+            end <- tail(table, 1)
+        } else {
+            beg <- tail(table, 1)
+            end <- head(table, 1)
```

Comment:
This is the same as the discussion above: the API returns selected rows in the order they appear in Crunch always, so we were always getting out of order results if the row numbers weren't ranges (eg if we `ds[c(12, 10, 3)]` we would get back `ds[c(3, 10, 12)]`.